src/ucp/wireup/wireup_cm.c

```diff
@@ -370,6 +368,7 @@ static void ucp_cm_client_connect_cb(uct_ep_h uct_cm_ep, void *arg,
     status      = connect_args->status;
 
     if (status != UCS_OK) {
+        ucp_ep->flags &= ~UCP_EP_FLAG_LOCAL_CONNECTED;
```

Comment:
same question - why in some cases we remove UCP_EP_FLAG_LOCAL_CONNECTED before calling set_ep_failed, and in others don't?

Reply:
we remove this flag is wireup is not completed - connection establishment failed.
if one from EPs has completed connection - other side has to call disconect.